### PR TITLE
🚀 Add release and publish process

### PR DESCRIPTION
**Pull request by bdeneux**
### 🚀 Release and Publish on crates.io

- Add workflow to release the project and binary
- Add workflow to publish on crates.io : 
  - We need to set `CARGO_API_TOKEN` secret into organization after Bot-anik (I guess) subscription into [crates.io](https://crates.io).

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Publish
+
+on:
+  push:
+    branches: [ main ]
```
Comment (ad2ien) [inline .github/workflows/publish.yml:5]: If I'm right, the `if` below ensures that publish is done only if triggered by a tag push, maybe we can only do 
```
on:
  push:
    tags: [ 'v*' ]
```

**Review by ad2ien — COMMENTED** [review]
Cool :dark_sunglasses:  !

:point_up: I've a remark

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ccamel) [reply .github/workflows/publish.yml:5]: Yes. But this is actually a general pattern. From a CI/CD perspective, the publishing workflow can be triggered by either a tag, a push to main or a pull_request. If you add a new job here, you won't have to worry about setting more triggers.

@amimart your opinion.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up context
+        id: project_context
+        uses: FranzDiebold/github-env-vars-action@v2.3.1
+
```
Comment (ccamel) [inline .github/workflows/publish.yml:26]: Is it actually used?

Comment (bdeneux) [reply .github/workflows/publish.yml]: Ah yes, I thought it was necessary for the environment variable but I misunderstood the purpose of this action. Thanks.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (ad2ien) [reply .github/workflows/publish.yml:5]: Ah yes so we can leave the triggers for the sake of consistency between the CIs.
Thanks

**Review by ad2ien — APPROVED** [review]

**Review by ccamel — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (amimart) [reply .github/workflows/publish.yml:5]: At least, we could remove the pull request trigger as we don't use it anymore on publish workflows.

**Review by hleonardtek — APPROVED** [review]

**Review by amimart — APPROVED** [review]
Very good ! 👍

**Comment by ccamel** [comment]
FYI @amimart @bdeneux I've set the `CARGO_API_TOKEN` secret at the organization level with the generated [crates.io](https://crates.io/) API key under the @bot-anik account.

**Comment by nperon** [comment]
👍 👍 👍

**Review by nperon — APPROVED** [review]

**Comment by bdeneux** [comment]
> FYI @amimart @bdeneux I've set the `CARGO_API_TOKEN` secret at the organization level with the generated [crates.io](https://crates.io/) API key under the @bot-anik account.

Thanks ! Let's go merge it 🚀.

**Pull request merged**